src/compute/src/logging/compute.rs

```diff
@@ -253,20 +256,29 @@ pub fn construct<A: Allocate>(
                                     if let Some(import_map) =
                                         dataflow_imports.get_mut(&dataflow_key)
                                     {
-                                        for (import_id, (time_deque, delay_map)) in import_map {
+                                        for (
+                                            import_id,
+                                            ImportDelayData {
+                                                time_deque,
+                                                delay_map,
+                                            },
+                                        ) in import_map
+                                        {
                                             while let Some(current_front) = time_deque.pop_front() {
                                                 let import_logical = current_front.0;
                                                 if logical >= import_logical {
                                                     let elapsed_ns =
                                                         time.as_nanos() - current_front.1;
-                                                    let delay_count =
-                                                        delay_map.entry(elapsed_ns).or_insert(0);
-                                                    *delay_count += 1;
                                                     let elapsed_pow =
                                                         elapsed_ns.next_power_of_two();
                                                     let elapsed_ns: i128 = elapsed_ns
                                                         .try_into()
                                                         .expect("elapsed_ns too big");
+                                                    let (delay_sum, delay_count) = delay_map
+                                                        .entry(elapsed_pow)
+                                                        .or_insert((0, 0));
+                                                    *delay_sum += elapsed_ns;
+                                                    *delay_count += 1;
```

Comment:
Computing these here seems to be a bit redundant with later arranging and performing a `count_total_core`. Perhaps you could just retract the entire aggregate here and emit a new one? That way, you could get rid of the intermediate arrangement downstream?

AFAIR, your initial version had an `explode_one` downstream, so the aggregation was delegated to DD. Now, the aggregation has to be done in two places. Perhaps it would avoid future inconsistencies to keep it only in one place, directly in the the demux operator?